firstvoices/backend/views/immersion_label_views.py

```diff
@@ -0,0 +1,155 @@
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    update=extend_schema(
+        description="Edit an immersion label in the specified site.",
+        responses={
+            200: OpenApiResponse(
+                description=doc_strings.success_200_edit,
+                response=ImmersionLabelDetailSerializer,
+            ),
+            400: OpenApiResponse(description=doc_strings.error_400_validation),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[
+            site_slug_parameter,
+            key_parameter,
+        ],
+    ),
+    partial_update=extend_schema(
+        description="Edit an immersion label in the specified site. Any omitted fields will be unchanged.",
+        responses={
+            200: OpenApiResponse(
+                description=doc_strings.success_200_edit,
+                response=ImmersionLabelDetailSerializer,
+            ),
+            400: OpenApiResponse(description=doc_strings.error_400_validation),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[
+            site_slug_parameter,
+            key_parameter,
+        ],
+    ),
+    destroy=extend_schema(
+        description="Delete an immersion label from the specified site.",
+        responses={
+            204: OpenApiResponse(description=doc_strings.success_204_deleted),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[site_slug_parameter, key_parameter],
+    ),
+    all=extend_schema(
+        description="Returns a mapping of immersion label keys to their corresponding dictionary entry titles.",
+        responses={
+            200: inline_serializer(
+                name="ImmersionLabelMap",
+                fields={
+                    "key": serializers.CharField(),
+                    "dictionary_entry_title": serializers.CharField(),
+                },
+            ),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+)
+class ImmersionLabelViewSet(
+    SiteContentViewSetMixin, FVPermissionViewSetMixin, ModelViewSet
+):
+    """
+    API endpoint that allows immersion labels to be viewed and edited.
+    """
+
+    lookup_field = "key"
+    serializer_class = ImmersionLabelDetailSerializer
+    permission_type_map = {
+        **FVPermissionViewSetMixin.permission_type_map,
+        "all": None,
```

Comment:
Why "None" for the permission here?

Reply:
I am matching the `all` endpoint to the list permission endpoint which also returns None